plugins/inputs/system/ps.go

```diff
@@ -99,21 +101,26 @@ func (s *systemPS) DiskUsage(
 			}
 		}
 
+		// If there's a host mount prefix, exclude other paths
+		if len(hostMountPrefix) > 0 && !strings.HasPrefix(p.Mountpoint, hostMountPrefix) {
```

Comment:
Then maybe rename the option to something more explicit like `HOST_MOUNT_PREFIX_FILTER` or something like that. From the user perspective you are changing the behavior of this option since it did not filter previously IIRC.
Feel free to discard my comment if that is not the case.